src/RustyObjectStore.jl

```diff
@@ -81,52 +98,203 @@ function init_object_store(config::ObjectStoreConfig=DEFAULT_CONFIG)
     return nothing
 end
 
+"""
+    $TYPEDEF
+
+# Keyword Arguments
+- `request_timeout_secs::Option{Int}`: (Optional) Client request timeout in seconds.
+- `connect_timeout_secs::Option{Int}`: (Optional) Client connection timeout in seconds.
+- `max_retries::Option{Int}`: (Optional) Maximum number of retry attempts.
+- `retry_timeout_secs::Option{Int}`: (Optional) Maximum amount of time from the initial request after which no further retries will be attempted (in seconds).
+"""
+struct ClientOptions
+    params::Dict{String, String}
+
+    function ClientOptions(;
+        request_timeout_secs::Option{Int} = nothing,
+        connect_timeout_secs::Option{Int} = nothing,
+        max_retries::Option{Int} = nothing,
+        retry_timeout_secs::Option{Int} = nothing
+    )
+        params = Dict()
+        if !isnothing(request_timeout_secs)
+            params["timeout"] = "$(string(request_timeout_secs))s"
+        end
+
+        if !isnothing(connect_timeout_secs)
+            params["connect_timeout"] = "$(string(connect_timeout_secs))s"
+        end
+
+        if !isnothing(max_retries)
+            params["max_retries"] = string(max_retries)
+        end
+
+        if !isnothing(retry_timeout_secs)
+            params["retry_timeout_secs"] = string(retry_timeout_secs)
+        end
+
+        return new(params)
+    end
+end
+
+abstract type AsConfig end
+
+function as_config(wrapper::AsConfig) end
+
 """
     $TYPEDEF
 
 # Arguments
-$TYPEDFIELDS
+- `url::String`: Url of the object store container root path.
+  It must include the cloud specific url scheme (s3://, azure:// ...).
+- `params::Dict{String, String}`: A set of key-value pairs to configure access to the object store.
+  Refer to the object_store crate documentation for the list of all supported parameters.
 """
-struct AzureCredentials
-    "Azure account name"
-    account::String
-    "Azure container name"
-    container::String
-    "Azure access key"
-    key::String
-    "(Optional) Alternative Azure host. For example, if using Azurite."
-    host::String
-    function AzureCredentials(account::String, container::String, key::String, host::String="")
-        return new(account, container, key, host)
+struct Config <: AsConfig
+    config_string::String
+    function Config(url::String, params::Dict{String, String})
+        dict = merge(Dict("url" => url), params)
+        config_string = JSON3.write(dict)
+        new(config_string)
     end
 end
-function Base.show(io::IO, credentials::AzureCredentials)
-    print(io, "AzureCredentials("),
-    print(io, repr(credentials.account), )
-    print(io, ", ", repr(credentials.container))
-    print(io, ", ", "\"*****\"") # don't print the secret key
-    !isempty(credentials.host) && print(io, ", ", repr(credentials.host))
+
+as_config(wrapper::Config) = wrapper
+
+"""
+    $TYPEDEF
+
```

Comment:
docstring should mention the need to provide exactly one of `storage_account_key` or `storage_sas_token`

Reply:
Will do!